src/azul/plugins/repository/tdr_anvil/__init__.py

```diff
@@ -940,17 +952,20 @@ def _retrieve_entities(self,
         else:
             return []
 
-    _schema_columns = {
-        table['name']: [column['name'] for column in table['columns']]
-        for table in anvil_schema['tables']
-    }
-
-    def _columns(self, table_name: str) -> set[str]:
-        try:
-            columns = self._schema_columns[table_name]
-        except KeyError:
-            return {'*'}
-        else:
-            columns = set(columns)
-            columns.add('datarepo_row_id')
-            return columns
+    @cached_property
+    def _columns_by_table(self) -> Mapping[str, AbstractSet[str]]:
```

Comment:
```suggestion
    def __columns_by_table(self) -> Mapping[str, AbstractSet[str]]:
```

unless the resulting name mangling breaks the caching.

Reply:
It *does* break the caching.
```
>>> from azul.azulclient import AzulClient
>>> p = AzulClient().repository_plugin('anvil')
'_Plugin__schema_columns_by_table'?                                                                                                               
>>> s1 = p._Plugin__schema_columns_by_table
>>> s2 = p._Plugin__schema_columns_by_table
>>> s1 == s2
True
>>> s1 is s2
False
```